cmd/root/bundle.go

```diff
@@ -108,19 +116,27 @@ func TryConfigureBundle(cmd *cobra.Command, args []string) error {
 	return configureBundle(cmd, args, bundle.TryLoad)
 }
 
-// environmentCompletion executes to autocomplete the argument to the environment flag.
-func environmentCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+// targetCompletion executes to autocomplete the argument to the target flag.
+func targetCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 	b, err := loadBundle(cmd, args, bundle.MustLoad)
 	if err != nil {
 		cobra.CompErrorln(err.Error())
 		return nil, cobra.ShellCompDirectiveError
 	}
 
-	return maps.Keys(b.Config.Environments), cobra.ShellCompDirectiveDefault
+	return maps.Keys(b.Config.Targets), cobra.ShellCompDirectiveDefault
+}
+
+func initTargetFlag(cmd *cobra.Command) {
+	// To operate in the context of a bundle, all commands must take an "target" parameter.
+	cmd.PersistentFlags().String("target", "", "bundle target to use (if applicable)")
+	cmd.RegisterFlagCompletionFunc("target", targetCompletion)
```

Comment:
Can you double check if we can safely use the `-t` shorthand as well?

Reply:
@pietern ”-t” is reserved for tokens flag at the moment, any suggestions for other shorthand? like -tt?

I also considered to continue use -e afterwards but it might be confusing

Reply:
nvm, it seems to be available and not used by other commands